File: python/Programmers/level1/lotto.py
def solution(lottos, win_nums):
    answer = []
    dic = {}
    high, low = 0, 0
    for win in win_nums:
        if win not in dic:
            dic[win] = 1

    for l in lottos:
        if l == 0:
            high += 1
        elif l in dic:
            high += 1
            low += 1

    score = {6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6}
    answer = [score[high], score[low]]
    return answer

File: java/Programmers/level1/Lotto.java
package Programmers.level1;

import java.util.*;

public class Lotto {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {0, 0};
        int max_count = 0;
        int min_count = 0;

        for (int i =0; i<lottos.length; i++){
            if(lottos[i] == 0){
                max_count++;
                continue;
            }

            for(int j=0; j<win_nums.length; j++){
                if(win_nums[j] == lottos[i]){
                    max_count++;
                    min_count++;
                }
            }
        }

        Map<Integer, Integer> score = new HashMap<>();
        for(int i=1; i<7; i++){
            score.put(7-i, i);
        }
        score.put(0, 6);

        answer[0] = score.get(max_count);
        answer[1] = score.get(min_count);

        return answer;
    }
}